addons/mail/models/mail_thread.py

```diff
@@ -2171,7 +2171,12 @@ def _notify_record_by_inbox(self, message, recipients_data, msg_vals=False, **kw
                 for partner_id in inbox_pids:
                     bus_notifications.append([(self._cr.dbname, 'ir.needaction', partner_id), dict(message_format_values)])
             if channel_ids:
-                bus_notifications += self.env['mail.channel'].sudo().browse(channel_ids)._channel_message_notifications(message, message_format_values)
+                channels = self.env['mail.channel'].sudo().browse(channel_ids)
+                bus_notifications += channels._channel_message_notifications(message, message_format_values)
+                for channel in channels.filtered(lambda x: x.email_send):
+                    for partner in channel.channel_partner_ids.filtered(lambda x: x.user_ids.notification_type == 'email'):
+                        channel.channel_seen(partner.id)
```

Comment:
Not a python expert, but I think it's best not change the signature of `channel_seen` and instead use `with_env`.
(Especially when this is a simplification that will likely change in the _near_ future?)